### PR TITLE
fix: update naming to fit network endpoints naming

### PR DESCRIPTION
**Pull request by julianiff**
-> naming update to reflect updates to the domain model of the [remote Service](https://www.figma.com/file/HVPqFN8juop6itsIFC2EWl/T1-2024-(January---April)?type=whiteboard&node-id=2262%3A11161&t=ptcR0p5CukIqofQs-1).

### CODE REVIEW

**Review by imoutaharik — APPROVED** [review]

File: packages/cli/src/utils/verifyManifest.ts [hunk 1]
```diff
@@ -26,24 +26,25 @@ const secretSchema = object({
 });
 const secretsArraySchema = array(secretSchema);
 
-const proxyOptionsSchema = object({
+const requestOptionsSchema = object({
     method: z.enum(['GET', 'POST', 'PUT', 'DELETE']),
-    headers: z.record(string()),
-    body: z.any(),
+    headers: z.record(string()).optional(),
+    body: z.any().optional(),
```
Comment (SamCreasey) [inline packages/cli/src/utils/verifyManifest.ts:32]: Maybe add a test case to cover these two new features? I don't think there's one for not throwing an error on a request that has no headers or body

Comment (julianiff) [reply packages/cli/src/utils/verifyManifest.ts:32]: Added additional test case

File: packages/cli/tests/utils/verifyManifest.spec.ts [hunk 2]
```diff
@@ -481,50 +482,53 @@ describe('Verify Platform App Manifest', () => {
...
+    it('should accept an array of network endpoint objects', async () => {
+        const verifiedManifest = await verifyManifest(MANIFEST_WITH_NETWORK_CALL, platformAppManifestSchemaV1);
         expect(!!verifiedManifest).toBe(true);
     });
 
-    it('should throw error when proxyCallNetwork is not an array', async () => {
-        await expect(
-            async () => await verifyManifest(MANIFEST_WITH_NOT_ARRAY_PROXY, platformAppManifestSchemaV1),
-        ).rejects.toThrow();
+    it('should accept an array of network endpoint without header and b ody', async () => {
```
Comment (SamCreasey) [inline packages/cli/tests/utils/verifyManifest.spec.ts:490]:
```suggestion
    it('should accept an array of network endpoint without header and body', async () => {
```

**Review by SamCreasey — APPROVED** [review]
1 typo but approved, nice! :)

**Review by mike85 — APPROVED** [review]
LGTM

**Pull request merged**